rules/wow.ts

```diff
@@ -0,0 +1,15 @@
+import * as YAML from 'yaml';
+
+import { defineRule } from '../api/api.ts';
+import { LOCATION } from '../core/rule.state.ts';
+
+export default defineRule({
+	id: 'wow',
+	progress: async ({ stаte }) => {
+		const totalProgresses = stаte?.totalProgresses || 0;
+		const state = { totalProgresses: totalProgresses + 1, ...stаte };
+
+		await Deno.writeTextFile(LOCATION, YAML.stringify(state));
+		console.log('wow');
+	},
+});
```

Comment:
An idea I had was that we could allow mutations to happen in rules to core state:

```suggestion
import { defineRule } from '../api/api.ts';

export default defineRule({
  id: 'wow',
  progress: async ({ core }) => {
    core.turnIndex = core?.turnIndex || 0;
  },
});
```

We'd need to change `core/rule.state.ts` to include:
```ts
export const STATE = z.object({
  id: z.string(),
  players: z.object({
    list: z.array(z.string()),
    active: z.string(),
  }),
  turnIndex: z.number(),
});
```

And then core-state will only get written to the file at the very end, somehow.